File: leetcode0005/solution.rs
impl Solution {
    pub fn longest_palindrome(s: String) -> String {
        let chars: Vec<char> = s.chars().collect();
        let len = chars.len();
        
        if len < 2 {
            return s;
        }
        
        let mut start = 0;
        let mut max_len = 1;
        
        for i in 0..len {
            let mut left = i as i32;
            let mut right = i as i32;
            
            // Expand around the center
            while right < len as i32 - 1 && chars[right as usize] == chars[(right + 1) as usize] {
                right += 1;
            }
            
            while left > 0 && right < len as i32 - 1 && chars[(left - 1) as usize] == chars[(right + 1) as usize] {
                left -= 1;
                right += 1;
            }
            
            let curr_len = right - left + 1;
            
            if curr_len > max_len {
                start = left as usize;
                max_len = curr_len;
            }
        }
        
        s[start..start+(max_len as usize)].to_string()
    }
}

File: leetcode0005/solution.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        if s == s[::-1]: return s
        
        start, size = 0, 1
        for i in range(1, len(s)):
            l, r = i - size, i + 1
            s1, s2 = s[l-1:r], s[l:r]
            print(s1, s2, f"\t{s[start: start + size]}")
            
            if l - 1 >= 0 and s1 == s1[::-1]:
                start, size = l - 1, size + 2
            elif s2 == s2[::-1]:
                start, size = l, size + 1
                
        return s[start:start+size]
# class Solution:
#     def longestPalindrome(self, s: str) -> str:
#         ans = ""
#         n = len(s)

#         if n == 1:
#             return s

#         def expand(left, right):
#             while left >= 0 and right < len(s) and s[left] == s[right]:
#                 left -= 1
#                 right += 1
#             return s[left + 1 : right]

#         for i in range(n):
#             tmp = expand(i, i)
#             if len(tmp) > len(ans):
#                 ans = tmp
#             tmp = expand(i, i + 1)
#             if len(tmp) > len(ans):
#                 ans = tmp

#         return ans